datastruct/seqptrmap.c

```diff
@@ -57,7 +57,7 @@ seqptrmap_add(struct seqptrmap * M, void * ptr)
 	M->len += 1;
 
```

Comment:
How about adding an assertion here to check that we don't overflow int64_t when we compute the value to return?  Being careful to not have an integer overflow while checking the assertion, of course...